# Market Returns and Volatility Analysis

In [85]:
import yfinance as yf
import pandas as pd
import numpy as np
from scipy.stats import norm
import plotly.express as px
import plotly.graph_objects as go

## 1. Data Collection

In [86]:
# Descargar los datos
sp500 = yf.download("^GSPC", start="2018-01-01")
sp500.head()

[*********************100%***********************]  1 of 1 completed


Price,Close,High,Low,Open,Volume
Ticker,^GSPC,^GSPC,^GSPC,^GSPC,^GSPC
Date,,,,,
2018-01-02,2695.810059,2695.889893,2682.360107,2683.729980,3397430000
2018-01-03,2713.060059,2714.370117,2697.770020,2697.850098,3544030000
2018-01-04,2723.989990,2729.290039,2719.070068,2719.310059,3697340000
2018-01-05,2743.149902,2743.449951,2727.919922,2731.330078,3239280000
2018-01-08,2747.709961,2748.510010,2737.600098,2742.669922,3246160000


In [87]:
# Convertir la serie a DataFrame
df_close=sp500[("Close","^GSPC")].to_frame()
df_close.head()

,Close
,^GSPC
Date,
2018-01-02,2695.810059
2018-01-03,2713.060059
2018-01-04,2723.989990
2018-01-05,2743.149902
2018-01-08,2747.709961


In [88]:
print(type(df_close))
print(df_close.columns)

<class 'pandas.DataFrame'>
MultiIndex([('Close', '^GSPC')],
           )


In [89]:
# Quitar el MultiIndex
df_close.columns = ['Close']
df_close.head()

,Close
Date,
2018-01-02,2695.810059
2018-01-03,2713.060059
2018-01-04,2723.989990
2018-01-05,2743.149902
2018-01-08,2747.709961


In [90]:
# Verificar valores faltantes
df_close.isna().sum()

Close    0
dtype: int64

## 2. S&P 500 Analysis

### 2.1 Price Evolution

In [91]:
# Gráfica del precio de cierre
close_graph = px.line(x = df_close.index, y = df_close['Close'])
close_graph.update_layout(title={'text': 'S&P 500 Closing Price', 'x':0.5},template='simple_white')
close_graph.update_xaxes(title_text='Date')
close_graph.update_yaxes(title_text='Price')

close_graph.update_xaxes(title_text="Return Rate", title_font=dict(size=15, color='#333333'),  
                         showgrid=True, gridcolor='#f0f0f0', zeroline=True, zerolinecolor='black')
close_graph.update_yaxes(title_text="Probability Density", title_font=dict(size=15, color='#333333'), 
                         showgrid=True, gridcolor='#f0f0f0')



### Interpretation

The S&P 500 increased substantially over the sample period, although its upward trend was interrupted by several declines. The price chart shows its overall evolution, while returns and drawdowns help assess the magnitude of daily fluctuations and losses from previous peaks.

### 2.2 Daily Returns

In [92]:
# Rendimientos
df_close['Simple_Return'] = (df_close['Close'] - df_close['Close'].shift())/df_close['Close'].shift()
df_close.head(5)

,Close,Simple_Return
Date,,
2018-01-02,2695.810059,NaN
2018-01-03,2713.060059,0.006399
2018-01-04,2723.989990,0.004029
2018-01-05,2743.149902,0.007034
2018-01-08,2747.709961,0.001662


In [93]:
# Log-rendimientos
df_close['Log_Return'] = np.log(df_close['Close']/df_close['Close'].shift())
df_close.head(5)

,Close,Simple_Return,Log_Return
Date,,,
2018-01-02,2695.810059,NaN,NaN
2018-01-03,2713.060059,0.006399,0.006378
2018-01-04,2723.989990,0.004029,0.004021
2018-01-05,2743.149902,0.007034,0.007009
2018-01-08,2747.709961,0.001662,0.001661


In [94]:
# Gráfica el Rendimiento Simple
return_graph = px.line(x = df_close.index, y = df_close['Simple_Return'])
return_graph.update_layout(title={'text': 'S&P 500 Simple Return', 'x':0.5},template='simple_white')
return_graph.update_xaxes(title_text='Date')
return_graph.update_yaxes(title_text='Return')

return_graph.update_xaxes(title_text="Return Rate", title_font=dict(size=15, color='#333333'),  
                         showgrid=True, gridcolor='#f0f0f0', zeroline=True, zerolinecolor='black')
return_graph.update_yaxes(title_text="Probability Density", title_font=dict(size=15, color='#333333'), 
                         showgrid=True, gridcolor='#f0f0f0')

In [95]:
# Gráfica el Log-Rendimiento
logreturn_graph = px.line(x = df_close.index, y = df_close['Log_Return'])
logreturn_graph.update_layout(title={'text': 'S&P 500 Log-Return', 'x':0.5},template='simple_white')
logreturn_graph.update_xaxes(title_text='Date')
logreturn_graph.update_yaxes(title_text='Return')

logreturn_graph.update_xaxes(title_text="Return Rate", title_font=dict(size=15, color='#333333'),  
                         showgrid=True, gridcolor='#f0f0f0', zeroline=True, zerolinecolor='black')
logreturn_graph.update_yaxes(title_text="Probability Density", title_font=dict(size=15, color='#333333'), 
                         showgrid=True, gridcolor='#f0f0f0')

### Interpretation

We can see that returns fluctuate around a more stable level than prices, although their variability changes over time. This makes returns useful for statistical analysis, but the graphs alone do not establish that they are stationary.

### 2.3 Descriptive Statistics

In [96]:
print('Simple Return Mean:', round(df_close['Simple_Return'].mean(),5))
print('Simple Return Median:', round(df_close['Simple_Return'].median(),5))
print('Simple Return Standard Deviation:', round(df_close['Simple_Return'].std(),5))
print('Simple Return Minimum:', round(df_close['Simple_Return'].min(),5))
print('Simple Return Maximum:', round(df_close['Simple_Return'].max(),5))
print('Simple Return Skewness:', round(df_close['Simple_Return'].skew(),5))
print('Simple Return Kurtosis:', round(df_close['Simple_Return'].kurtosis(),5))

Simple Return Mean: 0.00055
Simple Return Median: 0.00087
Simple Return Standard Deviation: 0.01206
Simple Return Minimum: -0.11984
Simple Return Maximum: 0.09515
Simple Return Skewness: -0.34977
Simple Return Kurtosis: 14.18318


In [97]:
df_close['Simple_Return'].describe()

count    2196.000000
mean        0.000548
std         0.012056
min        -0.119841
25%        -0.004419
50%         0.000871
75%         0.006466
max         0.095154
Name: Simple_Return, dtype: float64

### Interpretation

- The average daily return is approximately 0.055%.

- The median daily return, at 0.087%, is higher than the mean, which is consistent with the negative skewness observed in the sample.

- The standard deviation is approximately 1.21 percentage points, measuring the dispersion of daily returns around their mean.

- The minimum observed daily return was −11.98%, while the maximum was 9.52%.

- The skewness of approximately −0.350 indicates negative asymmetry in the sample.

- The excess kurtosis of 14.18 indicates more pronounced extreme observations than would be expected under a normal distribution, whose excess kurtosis is zero.

### 2.4 Return Distribution

In [98]:
return_hist = px.histogram(df_close['Simple_Return'], histnorm= 'probability density',nbins=100)

return_hist.update_traces(
    marker_color="#3f1fb4",
    marker_line_color='black',
    marker_line_width=0.7
)

return_hist.update_layout(
    title_text="S&P 500 Simple Return Histogram",
    title_x=0.5,
    title_font=dict(size=18, color='#333333'),
    template='simple_white',
    showlegend=False,
    width=800,
    height=450
)

return_hist.update_xaxes(title_text="Return Rate", title_font=dict(size=15, color='#333333'),  
                         showgrid=True, gridcolor='#f0f0f0', zeroline=True, zerolinecolor='black')
return_hist.update_yaxes(title_text="Probability Density", title_font=dict(size=15, color='#333333'), 
                         showgrid=True, gridcolor='#f0f0f0')


In [99]:
mu = df_close['Simple_Return'].mean()
sigma = df_close['Simple_Return'].std()
min_sr = df_close['Simple_Return'].min()
max_sr = df_close['Simple_Return'].max()

In [100]:
x_sr = np.linspace(min_sr, max_sr, 2100)
y_sr = norm.pdf(x_sr, mu, sigma)

In [101]:
return_hist.add_trace(
    go.Scatter(
        x=x_sr, 
        y=y_sr, 
        mode='lines',
        name="Normal Distribution",
        line=dict(color='red', width=1.5)
    )
)

return_hist.update_layout(
    template="simple_white",
    showlegend=True,  
    legend=dict(yanchor="top", y=0.99, xanchor="left", x=0.01), 
    title_x=0.5,
)

### Interpretation

We can see that the return distribution has a more concentrated center and more pronounced extreme observations than a normal distribution with the same mean and standard deviation. Its negative skewness also indicates asymmetry toward the left. Together, these results suggest that a normal distribution does not adequately describe the returns, particularly in the tails.

### 2.5 Rolling Volatility

In this section, daily volatility is annualized using 252 trading days per year:

$$
\sigma_{\text{annualized}}=\sqrt{252},\sigma_{\text{daily}}
$$

Rolling volatility is calculated using the standard deviation of returns over the previous 30 trading observations and then annualized using the same convention.

In [102]:
annual_simplereturn_std = np.sqrt(252) * df_close['Simple_Return'].std()
print('Volatilidad:', annual_simplereturn_std)

Volatilidad: 0.19138554941640262


In [103]:
annual_simplereturn_rw_std = np.sqrt(252) * df_close['Simple_Return'].rolling(window=30).std()
annual_simplereturn_rw_std.head()

Date
2018-01-02   NaN
2018-01-03   NaN
2018-01-04   NaN
2018-01-05   NaN
2018-01-08   NaN
Name: Simple_Return, dtype: float64

In [104]:
annual_simplereturn_rw_std.tail()

Date
2026-09-23    0.100036
2026-09-24    0.099707
2026-09-25    0.098963
2026-09-28    0.101252
2026-09-29    0.100604
Name: Simple_Return, dtype: float64

In [105]:
# Gráfica de la volatilidad móvil de los rendimientos simples
annual_simplereturn_rw_std_graph = px.line(x = annual_simplereturn_rw_std.index, y = annual_simplereturn_rw_std)
annual_simplereturn_rw_std_graph.update_layout(title={'text': 'S&P 500 Annualized Volatility (30-Day Rolling)', 'x':0.5})
annual_simplereturn_rw_std_graph.update_xaxes(title_text='Date')
annual_simplereturn_rw_std_graph.update_yaxes(title_text='Volatility')

annual_simplereturn_rw_std_graph.update_layout(
    title_font=dict(size=18, color='#333333'),
    template='simple_white',
    showlegend=False,
    width=1300,
    height=450
)

annual_simplereturn_rw_std_graph.update_xaxes(title_font=dict(size=15, color='#333333'),  
                         showgrid=True, gridcolor='#f0f0f0', zeroline=True, zerolinecolor='black')
annual_simplereturn_rw_std_graph.update_yaxes(title_font=dict(size=15, color='#333333'), 
                         showgrid=True, gridcolor='#f0f0f0')

We can observe a period of high volatility in the first half of 2020, coinciding with the onset of the COVID-19 crisis. Volatility increased again during 2022, against a backdrop of high inflation, rising interest rates, and Russia’s invasion of Ukraine. Another spike in the first half of 2025 coincided with U.S. tariff announcements. These events provide context, although the chart alone does not establish the causes of each increase.

The changing level of rolling volatility shows that fluctuations were concentrated in particular periods rather than remaining constant throughout the sample.

### 2.6 Drawdown

In [106]:
df_close['Running_Max']=df_close['Close'].cummax()
df_close['Drawdown']=(1-(df_close['Close'] / df_close['Running_Max']))*100
Maximum_Drawdown=df_close['Drawdown'].max()

In [107]:
df_close['Drawdown'].idxmax()

Timestamp('2020-03-23 00:00:00')

In [108]:
df_close[df_close['Drawdown'] == Maximum_Drawdown]

,Close,Simple_Return,Log_Return,Running_Max,Drawdown
Date,,,,,
2020-03-23,2237.399902,-0.029294,-0.029732,3386.149902,33.92496


In [109]:
df_close['Drawdown'].idxmax().date()

datetime.date(2020, 3, 23)

In [110]:
# Gráfica del Drawdown
drawdown_graph = px.line(x = df_close['Drawdown'].index, y = df_close['Drawdown'])
drawdown_graph.update_layout(title={'text': 'S&P 500 Drawdown', 'x':0.5})
drawdown_graph.update_xaxes(title_text='Date')
drawdown_graph.update_yaxes(title_text='Drawdown %')
drawdown_graph.update_traces(line_color="#080075")

drawdown_graph.update_layout(
    title_font=dict(size=18, color='#333333'),
    template='simple_white',
    showlegend=False,
    width=1500,
    height=450
)

drawdown_graph.update_xaxes(title_font=dict(size=15, color='#333333'),  
                         showgrid=True, gridcolor='#f0f0f0', zeroline=True, zerolinecolor='black')
drawdown_graph.update_yaxes(title_font=dict(size=15, color='#333333'), 
                         showgrid=True, gridcolor='#f0f0f0')

drawdown_graph.add_annotation(
    x=df_close['Drawdown'].idxmax(),
    y=Maximum_Drawdown,
    text=f"Maximum Drawdown<br>{Maximum_Drawdown.round(2)}%<br>{df_close['Drawdown'].idxmax().date()}",
    font=dict(color="#940000"),
    showarrow=True,
    arrowhead=2,
    arrowcolor="#940000",
    ax=-50,
    ay=-40
)

The largest observed drawdown was 33.92%, reached on March 23, 2020. This means that the S&P 500 was 33.92% below its previous highest closing level within the sample; it does not represent a one-day loss.

Drawdown is reported here as a positive loss magnitude. A value of zero indicates that the index is at its running peak, while larger values indicate deeper declines from that peak.

### 2.7 Annualized Volatility by Year

In [111]:
annual_volatility_by_year=np.sqrt(252) * df_close['Simple_Return'].groupby(df_close.index.year).std()
annual_volatility_by_year

Date
2018    0.170643
2019    0.124721
2020    0.344310
2021    0.131001
2022    0.241735
2023    0.130929
2024    0.126498
2025    0.187527
2026    0.131933
Name: Simple_Return, dtype: float64

In [112]:
#Gráfica de la volatilidad por año


annual_volatility_by_year_graph = px.bar(x = annual_volatility_by_year.index, y = annual_volatility_by_year, color_discrete_sequence=["#1f77b4"], text_auto=True)
annual_volatility_by_year_graph.update_layout(title={'text': 'Annualized Volatility of S&P 500 Daily Returns', 'x':0.5})
annual_volatility_by_year_graph.update_xaxes(title_text='Year')
annual_volatility_by_year_graph.update_yaxes(title_text='Annual Volatility', tickformat='.1%')

annual_volatility_by_year_graph.update_layout(
    title_font=dict(size=18, color='#333333'),
    template='simple_white',
    showlegend=False,
    width=1000,
    height=500
)

annual_volatility_by_year_graph.update_xaxes(title_font=dict(size=15, color='#333333'),  
                         showgrid=True, gridcolor='#f0f0f0', zeroline=True, zerolinecolor='black')
annual_volatility_by_year_graph.update_yaxes(title_font=dict(size=15, color='#333333'), 
                         showgrid=True, gridcolor='#f0f0f0')


Annualized volatility was highest in 2020, at 34.43%, followed by 2022, at 24.17%. These years experienced greater daily return fluctuations than the other years in the sample. By comparison, 2019 had the lowest annualized volatility, at 12.47%.

The 2026 estimate of 13.19% uses only observations available through September 29, so it should be treated as a partial-year estimate. These figures measure return variability, not whether the index gained or lost value during each year.

## 3. Multi-Asset Analysis

The multi-asset analysis compares the S&P 500, Nasdaq-100, gold, and intermediate-term U.S. Treasury bonds over a common sample period beginning in 2018.

Daily closing prices are used to compute simple returns. Risk is evaluated through annualized volatility, maximum drawdown, the Sharpe ratio, and historical Value at Risk (VaR). To ensure comparability across assets, only dates with available observations for all selected instruments are retained.

Annualized return is estimated from the arithmetic mean of daily returns multiplied by 252 trading days, while annualized volatility is calculated as the daily standard deviation multiplied by the square root of 252. The Sharpe ratio assumes a constant annual risk-free rate of 4%. Historical VaR is estimated empirically from the 95th percentile of daily losses.

### 3.1 Asset Selection and Data Preparation

In [113]:
tickers = ["^GSPC", "^NDX", "GC=F", "IEF"]
multi_asset_data = yf.download(tickers, start="2018-01-01")

multi_asset_data.head()

[*********************100%***********************]  4 of 4 completed


Price             Close                                              High  \
Ticker             GC=F        IEF        ^GSPC         ^NDX         GC=F   
Date                                                                        
2018-01-02  1313.699951  85.227943  2695.810059  6511.339844  1317.599976   
2018-01-03  1316.199951  85.317024  2713.060059  6575.799805  1320.099976   
2018-01-04  1319.400024  85.276550  2723.989990  6584.580078  1322.000000   
2018-01-05  1320.300049  85.171265  2743.149902  6653.290039  1320.300049   
2018-01-08  1318.599976  85.130768  2747.709961  6676.629883  1319.900024   

Price                                                    Low             \
Ticker            IEF        ^GSPC         ^NDX         GC=F        IEF   
Date                                                                      
2018-01-02  85.430443  2695.889893  6513.020020  1302.300049  85.122640   
2018-01-03  85.381826  2714.370117  6579.250000  1312.099976  85.227924   
2018-01-04  85.284652  2729.290039  6603.169922  1319.400024  85.065949   
2018-01-05  85.308963  2743.449951  6654.979980  1320.300049  85.098362   
2018-01-08  85.227970  2748.510010  6680.319824  1314.599976  85.057871   

Price                                        Open                          \
Ticker            ^GSPC         ^NDX         GC=F        IEF        ^GSPC   
Date                                                                        
2018-01-02  2682.360107  6417.750000  1302.300049  85.406144  2683.729980   
2018-01-03  2697.770020  6519.259766  1320.000000  85.333221  2697.850098   
2018-01-04  2719.070068  6576.919922  1319.400024  85.130750  2719.310059   
2018-01-05  2727.919922  6605.029785  1320.300049  85.276562  2731.330078   
2018-01-08  2737.600098  6649.200195  1319.900024  85.211766  2742.669922   

Price                   Volume                                         
Ticker             ^NDX   GC=F        IEF         ^GSPC          ^NDX  
Date                                                                   
2018-01-02  6431.589844   68.0  3902000.0  3.397430e+09  1.929700e+09  
2018-01-03  6520.029785   42.0  2135900.0  3.544030e+09  2.173130e+09  
2018-01-04  6595.750000    2.0  2430000.0  3.697340e+09  2.103220e+09  
2018-01-05  6613.129883    1.0  2163600.0  3.239280e+09  2.024000e+09  
2018-01-08  6651.680176   41.0  2450600.0  3.246160e+09  2.055010e+09

In [114]:
multi_asset_data.columns

MultiIndex([( 'Close',  'GC=F'),
            ( 'Close',   'IEF'),
            ( 'Close', '^GSPC'),
            ( 'Close',  '^NDX'),
            (  'High',  'GC=F'),
            (  'High',   'IEF'),
            (  'High', '^GSPC'),
            (  'High',  '^NDX'),
            (   'Low',  'GC=F'),
            (   'Low',   'IEF'),
            (   'Low', '^GSPC'),
            (   'Low',  '^NDX'),
            (  'Open',  'GC=F'),
            (  'Open',   'IEF'),
            (  'Open', '^GSPC'),
            (  'Open',  '^NDX'),
            ('Volume',  'GC=F'),
            ('Volume',   'IEF'),
            ('Volume', '^GSPC'),
            ('Volume',  '^NDX')],
           names=['Price', 'Ticker'])

In [115]:
multi_asset_close = multi_asset_data['Close']
multi_asset_close.head()


Ticker,GC=F,IEF,^GSPC,^NDX
Date,,,,
2018-01-02,1313.699951,85.227943,2695.810059,6511.339844
2018-01-03,1316.199951,85.317024,2713.060059,6575.799805
2018-01-04,1319.400024,85.276550,2723.989990,6584.580078
2018-01-05,1320.300049,85.171265,2743.149902,6653.290039
2018-01-08,1318.599976,85.130768,2747.709961,6676.629883


In [116]:
multi_asset_close.info()

<class 'pandas.DataFrame'>
DatetimeIndex: 2200 entries, 2018-01-02 to 2026-09-29
Data columns (total 4 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   GC=F    2198 non-null   float64
 1   IEF     2197 non-null   float64
 2   ^GSPC   2197 non-null   float64
 3   ^NDX    2197 non-null   float64
dtypes: float64(4)
memory usage: 85.9 KB


In [117]:
multi_asset_close = multi_asset_close.dropna()

multi_asset_return = multi_asset_close.pct_change().dropna()
multi_asset_return.head()

Ticker,GC=F,IEF,^GSPC,^NDX
Date,,,,
2018-01-03,0.001903,0.001045,0.006399,0.009900
2018-01-04,0.002431,-0.000474,0.004029,0.001335
2018-01-05,0.000682,-0.001235,0.007034,0.010435
2018-01-08,-0.001288,-0.000475,0.001662,0.003508
2018-01-09,-0.005233,-0.004757,0.001303,0.000196


### 3.2 Normalized Performance

In [118]:
normalized_prices = 100 * multi_asset_close/multi_asset_close.iloc[0]

asset_names = {"^GSPC": "S&P 500",
    "^NDX": "Nasdaq-100",
    "GC=F": "Gold",
    "IEF": "US Treasury Bonds"}

normalized_prices = normalized_prices.rename(columns=asset_names)

normalized_prices_fig = px.line(
    normalized_prices,
    x=normalized_prices.index,
    y=normalized_prices.columns,
    title="Normalized Performance of Selected Assets (2018–2026)"
)

normalized_prices_fig.update_layout(
    xaxis_title="Date",
    yaxis_title="Normalized Value (Start = 100)",
    legend_title="Asset",
    template="simple_white",
    width=1000,
    height=500
)

normalized_prices_fig.update_xaxes(title_font=dict(size=15, color='#333333'),  
                         showgrid=True, gridcolor='#f0f0f0', zeroline=True, zerolinecolor='black')
normalized_prices_fig.update_yaxes(title_font=dict(size=15, color='#333333'), 
                         showgrid=True, gridcolor='#f0f0f0')


Nasdaq-100 has the strongest cumulative performance, but we can see substantial declines, particularly around 2022 and 2025. The S&P 500 grows less dramatically, while gold behaves quite differently and accelerates strongly later in the sample. IEF is comparatively flat over the full period.

### 3.3 Risk and Return Comparison

In [119]:
annualized_return = multi_asset_return.mean() * 252
annualized_volatility = multi_asset_return.std() * np.sqrt(252)

In [120]:
# Maximum drawdown

running_max = multi_asset_close.cummax()
drawdown = 1- multi_asset_close/running_max
maximum_ma_drawdown = drawdown.max()

asset_comparison = pd.DataFrame({'Annualized Return': annualized_return,
                                 'Annualized Volatility': annualized_volatility,
                                 'Maximum Drawdown': maximum_ma_drawdown})

asset_comparison

,Annualized Return,Annualized Volatility,Maximum Drawdown
Ticker,,,
GC=F,0.148764,0.176386,0.249379
IEF,0.007675,0.068818,0.239248
^GSPC,0.138352,0.191531,0.339250
^NDX,0.205559,0.240361,0.355631


The Nasdaq-100 had the highest annualized arithmetic mean return, at 20.56%, but also the highest annualized volatility, at 24.04%, and the largest maximum drawdown, at 35.56%.

Gold had a higher annualized arithmetic mean return than the S&P 500 in this sample, at 14.88% versus 13.84%, together with lower volatility and a smaller maximum drawdown.

IEF had the lowest annualized volatility, at 6.88%, and the lowest annualized arithmetic mean return, at 0.77%. However, its maximum drawdown of 23.92% shows that relatively low daily volatility can still accompany a substantial cumulative decline.

These annualized returns are based on mean daily returns multiplied by 252; they are not compound annual growth rates.

### 3.4 Correlation Analysis

In [121]:
correlation_matrix = multi_asset_return.corr()
correlation_matrix = correlation_matrix.rename(index=asset_names, columns=asset_names)
correlation_matrix

Ticker,Gold,US Treasury Bonds,S&P 500,Nasdaq-100
Ticker,,,,
Gold,1.000000,0.244766,0.095581,0.099587
US Treasury Bonds,0.244766,1.000000,-0.103474,-0.057264
S&P 500,0.095581,-0.103474,1.000000,0.936494
Nasdaq-100,0.099587,-0.057264,0.936494,1.000000


In [122]:
correlation_matrix_fig = px.imshow(
    correlation_matrix,
    text_auto='.2f',
    aspect='auto',
    title='Correlation Matrix of Daily Returns',
    color_continuous_scale="RdBu_r",
    zmin=-1,
    zmax=1
)

correlation_matrix_fig.update_layout(
    template='simple_white',
    width=700,
    height=600
)

correlation_matrix_fig.update_xaxes(title_text='Asset')
correlation_matrix_fig.update_yaxes(title_text='Asset')



- S&P 500 vs Nasdaq-100: 0.94. Very strong positive correlation. They usually move in the same direction.
- Gold vs equities: 0.10. Very weak positive correlation.
- Treasury bonds vs S&P 500: -0.10 and vs Nasdaq-100: -0.06. Weak negative correlation.
- Gold vs Treasury bonds: 0.24. Modest positive correlation.

### 3.5 Sharpe Ratio

$$ \text{Sharpe Ratio} = \frac{E[R]-R_f}{\sigma} $$

For simplicity, this analysis assumes a constant annual risk-free rate of 4% over the full sample period.

In [123]:
risk_free_rate = 0.04
sharpe_ratio = (annualized_return - risk_free_rate)/annualized_volatility
asset_comparison["Sharpe Ratio"] = sharpe_ratio
asset_comparison

,Annualized Return,Annualized Volatility,Maximum Drawdown,Sharpe Ratio
Ticker,,,,
GC=F,0.148764,0.176386,0.249379,0.616629
IEF,0.007675,0.068818,0.239248,-0.469718
^GSPC,0.138352,0.191531,0.339250,0.513507
^NDX,0.205559,0.240361,0.355631,0.688794


Under the assumed constant annual risk-free rate of 4%, the Nasdaq-100 had the highest Sharpe ratio, at 0.69, followed by gold at 0.62 and the S&P 500 at 0.51. The Nasdaq-100 therefore had the highest estimated excess return per unit of volatility among the selected series during this sample.

IEF had a negative Sharpe ratio of −0.47 because its annualized mean return was below the assumed risk-free rate. This does not mean that its return was negative. These comparisons depend on the selected sample and the 4% assumption.

### 3.6 Historical Value at Risk

We define VaR as

$$\text{VaR}_p=F^{-1}(p)$$

where $F$ is the distribution of daily losses, $L = -R$.

In [124]:
multi_asset_loss = -multi_asset_return
historical_var_95 = multi_asset_loss.quantile(0.95)
asset_comparison["Historical VaR (95%) of Loss"] = historical_var_95
asset_comparison


,Annualized Return,Annualized Volatility,Maximum Drawdown,Sharpe Ratio,Historical VaR (95%) of Loss
Ticker,,,,,
GC=F,0.148764,0.176386,0.249379,0.616629,0.017107
IEF,0.007675,0.068818,0.239248,-0.469718,0.006861
^GSPC,0.138352,0.191531,0.339250,0.513507,0.017589
^NDX,0.205559,0.240361,0.355631,0.688794,0.024068


The Nasdaq-100 had the highest 95% one-day historical VaR, at 2.41%, followed by the S&P 500 at 1.76%, gold at 1.71%, and IEF at 0.69%.

For the Nasdaq-100, this means that approximately 5% of observed daily losses exceeded 2.41%. The same interpretation applies to the other series at their respective thresholds.

Historical VaR describes a threshold in the sample loss distribution. It is not the maximum possible loss, does not indicate how severe losses beyond the threshold may be, and does not guarantee the same frequency of exceedances in the future.

## 4. Conclusions

The analysis shows clear differences in both performance and risk across the selected assets.

The Nasdaq-100 generated the highest annualized return over the sample period, but it also exhibited the highest annualized volatility, the largest maximum drawdown, and the highest one-day historical VaR. This reflects a higher-risk, higher-return profile.

The S&P 500 also delivered strong returns, although with slightly lower volatility and risk than the Nasdaq-100. Its very high correlation with the Nasdaq-100 indicates that holding both provides relatively limited diversification benefits.

Gold showed a different return pattern and a much lower correlation with U.S. equities. Its lower correlation, combined with moderate volatility and drawdown, suggests that it may provide diversification benefits in a multi-asset portfolio.

U.S. Treasury bonds exhibited the lowest volatility among the selected assets, although their average return over the sample period was considerably lower. Under the assumed 4% constant risk-free rate, the bond ETF produced a negative Sharpe ratio, indicating that its return did not compensate for the assumed risk-free alternative during this period.

The correlation analysis highlights the importance of diversification across asset classes. While the S&P 500 and Nasdaq-100 moved very closely together, gold and Treasury bonds showed much weaker relationships with equities.

Finally, the comparison of volatility, maximum drawdown, Sharpe ratio, and historical VaR demonstrates that no single metric fully describes investment risk. Volatility measures the dispersion of returns, drawdown captures losses from previous peaks, the Sharpe ratio evaluates return relative to volatility, and historical VaR estimates a loss threshold based on the empirical distribution of past returns.